src/DoctrineObject.php

```diff
@@ -570,25 +572,33 @@ protected function handleTypeConversions($value, $typeOfField)
                 $value = (double) $value;
                 break;
             case 'datetimetz':
+            case 'datetimetz_immutable':
             case 'datetime':
+            case 'datetime_immutable':
             case 'time':
             case 'date':
                 if ($value === '') {
                     return null;
                 }
 
-                if ($value instanceof DateTime) {
+                if ($value instanceof DateTimeInterface) {
                     return $value;
                 }
 
+                $isImmutable = substr($typeOfField, -9) === 'immutable';
```

Comment:
```suggestion
                $isImmutable = substr($typeOfField, -9) === 'immutable';
                if (
                    ($isImmutable && $value instanceof DateTimeImmutable)
                    || (!$isImmutable && $value instanceof DateTime)
                ) {
                    return $value;
                } elseif ($isImmutable && $value instanceof DateTime) {
                    return DateTimeImmutable::createFromMutable($value);
                } elseif ($!isImmutable && $value instanceof DateTimeImmutable) {
                    return DateTime::createFromImmutable($value);
                }
```

I think this part needs to be more explicit, since for immutable fields the function should only DateTimeImmutable should be returned and for non-immutable fields only DateTime should be returned. Simply checking for DateTimeInterface will not guarantee this.

Reply:
DateTime::createFromImmutable is only available since PHP 7.3 - project is still targeting 7.2.

An alternative might be

`return new DateTime($value->format(DateTimeInterface::ISO8601));`

Thoughts?

Reply:
I think for 2.2.0 we should rather drop 7.2 anyways, as it is EOL for almost one year now. Dropping 7.2 is already pending in #29. Dropping PHP versions is usually considered minor changes and so are new features, so both could/must go in a new minor release anyways.